python/lsst/pipe/tasks/subtractBrightStars.py

```diff
@@ -0,0 +1,359 @@
+#
+"""Retrieve extended PSF model and subtract bright stars at calexp (ie
+single visit) level.
+"""
+
+__all__ = ["SubtractBrightStarsConnections", "SubtractBrightStarsConfig", "SubtractBrightStarsTask"]
+
+from functools import reduce
+from operator import ior
+
+import numpy as np
+from lsst.afw.image import Exposure, ExposureF, MaskedImageF
+from lsst.afw.math import (
+    StatisticsControl,
+    WarpingControl,
+    makeStatistics,
+    rotateImageBy90,
+    stringToStatisticsProperty,
+    warpImage,
+)
+from lsst.geom import Box2I, Point2D, Point2I
+from lsst.pex.config import ChoiceField, Field, ListField
+from lsst.pipe.base import PipelineTask, PipelineTaskConfig, PipelineTaskConnections, Struct
+from lsst.pipe.base import connectionTypes as cT
+
+
+class SubtractBrightStarsConnections(
+    PipelineTaskConnections,
+    dimensions=("instrument", "visit", "detector"),
+    defaultTemplates={"outputExposureName": "brightStar_subtracted", "outputBackgroundName": "brightStars"},
+):
+    inputExposure = cT.Input(
+        doc="Input exposure from which to subtract bright star stamps.",
+        name="calexp",
+        storageClass="ExposureF",
+        dimensions=(
+            "visit",
+            "detector",
+        ),
+    )
+    inputBrightStarStamps = cT.Input(
+        doc="Set of preprocessed postage stamps, each centered on a single bright star.",
+        name="brightStarStamps",
+        storageClass="BrightStarStamps",
+        dimensions=(
+            "visit",
+            "detector",
+        ),
+    )
+    inputExtendedPsf = cT.Input(
+        doc="Extended PSF model.",
+        name="extended_psf",
+        storageClass="ExtendedPsf",
+        dimensions=("band",),
+    )
+    skyCorr = cT.Input(
+        doc="Input Sky Correction to be subtracted from the calexp if ``doApplySkyCorr``=True.",
+        name="skyCorr",
+        storageClass="Background",
+        dimensions=(
+            "instrument",
+            "visit",
+            "detector",
+        ),
+    )
+    outputExposure = cT.Output(
+        doc="Exposure with bright stars subtracted.",
+        name="{outputExposureName}_calexp",
+        storageClass="ExposureF",
+        dimensions=(
+            "visit",
+            "detector",
+        ),
+    )
+    outputBackgroundExposure = cT.Output(
+        doc="Exposure containing only the modelled bright stars.",
+        name="{outputBackgroundName}_calexp_background",
+        storageClass="ExposureF",
+        dimensions=(
+            "visit",
+            "detector",
+        ),
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+        if not config.doApplySkyCorr:
+            self.inputs.remove("skyCorr")
+
+
+class SubtractBrightStarsConfig(PipelineTaskConfig, pipelineConnections=SubtractBrightStarsConnections):
+    """Configuration parameters for SubtractBrightStarsTask"""
+
+    doWriteSubtractor = Field(
```

Comment:
Either in this ticket, or in the refactoring ticket that should follow soon, it'd be good to make it as `Field[bool]` so type annotations can work.

Reply:
Type annotations added to all `Config` options (including `ChoiceField`, which I think works now?)